scripts/globals/spells/black/aspir.lua

```diff
@@ -52,6 +52,19 @@ spellObject.onSpellCast = function(caster, target, spell)
         target:delMP(dmg)
     end
 
+    -- Messaging Fixes
+    local currMP = caster:getMP()
+    local maxMP  = caster:getMaxMP()
+
+    if caster:getMP() == caster:getMaxMP() then
+        spell:setMsg(xi.msg.basic.MAGIC_DRAIN_MP, 0) -- Drains 0 MP
+        return 0
+    elseif dmg + currMP > maxMP then
+        local mpRecovered = maxMP - currMP
+        spell:setMsg(xi.msg.basic.MAGIC_DRAIN_MP, mpRecovered)
+        return 0
```

Comment:
Does this mean the spell will do 0 damage?

I would expect you could do something like:
```lua
 -- Messaging Fixes
local mpRecovered = caster:getMaxMP() - caster:getMP()

spell:setMsg(xi.msg.basic.MAGIC_DRAIN_MP, math.min(dmg, mpRecovered))

return dmg

Reply:
My current changes work appropriately. If player has 1 hp between currHP and max, it will report to the players the full amount drained, not 1 hp.
As far as I know. If I'm wrong let me know.

Reply:
I'm pretty sure Drain/Aspir will report the lower of whatever recovered and the total drain done.

HP: 9/10
Drain for 2HP
"You drain 1HP"

HP: 5/10
Drain for 2HP
"You drain 2HP"

Reply:
I'll test this on retail to determine proper behavior

Reply:
And the chatlog says you drained for 20? Weird, I always remember it being the min